pkg/controllers/clusteroperator_controller_test.go

```diff
@@ -404,6 +446,50 @@ var _ = Describe("Component sync controller", func() {
 			},
 			expectProvisioned: false,
 		}),
+		Entry("Should not provision resources because KCM still owns the controllers", testCase{
+			status: &configv1.InfrastructureStatus{
+				InfrastructureTopology: configv1.HighlyAvailableTopologyMode,
+				ControlPlaneTopology:   configv1.HighlyAvailableTopologyMode,
+				Platform:               configv1.AWSPlatformType,
+				PlatformStatus: &configv1.PlatformStatus{
+					Type: configv1.AWSPlatformType,
+				},
+			},
+			featureGateSpec: externalFeatureGateSpec,
+			kcmStatus: &operatorv1.KubeControllerManagerStatus{
+				StaticPodOperatorStatus: operatorv1.StaticPodOperatorStatus{
+					OperatorStatus: operatorv1.OperatorStatus{
+						Conditions: []operatorv1.OperatorCondition{
+							{
+								Type:               KCMCloudControllerOwnershipCondition,
+								Status:             operatorv1.ConditionTrue,
+								LastTransitionTime: metav1.Now(),
+							},
+						},
+					},
+				},
+			},
+			expectProvisioned: false,
+		}),
+		Entry("Should not provision resources because KCMO hasn't set CloudControllerOwner condition", testCase{
```

Comment:
How does this affect bootstrapping of the cluster? We need to verify that this won't break bootstrapping. IIUC KCMO will only start updating it's ClusterOperator after CCM has initialized the masters